### PR TITLE
fix(customs): Report errno to customs when password check fails.

### PR DESCRIPTION
**Pull request by rfk**
Otherwise we report the default errno of `999`, which isn't very helpful.

### CODE REVIEW

File: lib/routes/utils/password_check.js [hunk 1]
```diff
@@ -35,26 +43,29 @@ module.exports = function (log, config, Password, customs, db) {
...
-                  }
+          return customs.flag(clientAddress, {
+            email: emailRecord.email,
+            errno: error.ERRNO.INCORRECT_PASSWORD
+          })
+          .then(
+            function (result) {
+              if (result.lockout) {
+                log.info({
+                  op: 'account.lock',
```
Comment (vbudhram) [inline lib/routes/utils/password_check.js]: Would it makes sense to log a reason for lockout? In this case to many failed password checks.

**Comment by TDA** [comment]
I dig this! 👍

**Comment by vbudhram** [comment]
Minor nit, add some spies to [mockCustom](https://github.com/mozilla/fxa-auth-server/blob/master/test/local/password_check_test.js#L12) to ensure correct codes were sent. Otherwise, LGTM 👍

**Comment by rfk** [comment]
> add some spies to mockCustom to ensure correct codes were sent

Done.  Thanks @vbudhram, merging once travis gives the OK...

**Pull request merged**